kroxylicious-operator/src/test/java/io/kroxylicious/kubernetes/operator/KafkaProtocolFilterReconcilerIT.java

```diff
@@ -158,6 +159,46 @@ void shouldUpdateStatusOnSecretModify() {
         assertAllConditionsTrue(filterOne);
     }
 
+    @Test
+    void shouldUpdateReferentAnnotationOnSecretModify() {
+        // given
+        var filterOne = createFilterFirst();
+        String checksum = testActor.get(KafkaProtocolFilter.class, ResourcesUtil.name(filterOne)).getMetadata().getAnnotations()
+                .getOrDefault(MetadataChecksumGenerator.REFERENT_CHECKSUM_ANNOTATION, NO_CHECKSUM_SPECIFIED);
+        assertThat(checksum).isNotEqualTo(NO_CHECKSUM_SPECIFIED);
```

Comment:
call me old fashioned or a pedant but it feels very odd to see `assertThat` in the given block. One can use `assumeThat` if really needed, but generally we should be able to depend on the other tests to prevent this situation.

Reply:
yeah, I think the check might instead belong in the `then` of some of the other tests where the references resolve